base/essentials.jl

```diff
@@ -159,11 +164,44 @@ function typename(a::Union)
 end
 typename(union::UnionAll) = typename(union.body)
 
-convert(::Type{T}, x::Tuple{Any,Vararg{Any}}) where {T<:Tuple{Any,Vararg{Any}}} =
-    tuple(convert(tuple_type_head(T),x[1]), convert(tuple_type_tail(T), tail(x))...)
-convert(::Type{T}, x::T) where {T<:Tuple{Any,Vararg{Any}}} = x
-
-oftype(x,c) = convert(typeof(x),c)
+convert(::Type{T}, x::T) where {T<:Tuple{Any, Vararg{Any}}} = x
+convert(::Type{T}, x::Tuple{Any, Vararg{Any}}) where {T<:Tuple} =
+    (convert(tuple_type_head(T), x[1]), convert(tuple_type_tail(T), tail(x))...)
+
+# TODO: the following definitions are equivalent (behaviorally) to the above method
+# I think they may be faster / more efficient for inference,
+# if we could enable them, but are they?
+# TODO: These currently can't be used (#21026) since with
+#     z(::Type{Tuple{Val{T}} where T}) = T
```

Comment:
Indeed, I seem to have been tired when I tried to write the MRE. ref https://github.com/JuliaLang/julia/issues/23017

On master, this bug in the subtyping environment exhibits itself as a MethodError:
```julia
julia> z2(::Type{<:Tuple{Vararg{T}}}) where {T} = T

julia> z2(Tuple{Val{T}} where T)
Val{T}

julia> z2(Tuple{Val})
ERROR: MethodError: no method matching z2(::Type{Tuple{Val}})
```